### ISSUE TITLE
--set-string doesn't seem to handle booleans properly

### ISSUE THREAD

**Issue by koniiiik** [kubernetes/helm#4006]
Output of `helm version`:
```
Client: &version.Version{SemVer:"v2.9.0", GitCommit:"f6025bb9ee7daf9fee0026541c90a6f557a3e0bc", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.9.0", GitCommit:"f6025bb9ee7daf9fee0026541c90a6f557a3e0bc", GitTreeState:"clean"}
```

Output of `kubectl version`:
```
Client Version: version.Info{Major:"1", Minor:"10", GitVersion:"v1.10.2", GitCommit:"81753b10df112992bf51bbc2c2f85208aad78335", GitTreeState:"archive", BuildDate:"2018-04-30T07:09:42Z", GoVersion:"go1.10.1", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"8", GitVersion:"v1.8.12", GitCommit:"5d26aba6949f188fde1af4875661e038f538f2c6", GitTreeState:"clean", BuildDate:"2018-04-23T23:09:30Z", GoVersion:"go1.8.3", Compiler:"gc", Platform:"linux/amd64"}
```

Cloud Provider/Platform (AKS, GKE, Minikube etc.): 
AWS

When trying to install `stable/kubernetes-dashboard` with `nginx.ingress.kubernetes.io/secure-backends: "true"`as an annotation on the ingress, using `--set-string "ingress.annotations.nginx\.ingress\.kubernetes\.io/secure-backends=true"`, the following ends up in the resulting YAML:
```
# Source: kubernetes-dashboard/templates/ingress.yaml
apiVersion: extensions/v1beta1
kind: Ingress
metadata:
  name: kube-dashboard-kubernetes-dashboard
  labels:
    app: kubernetes-dashboard
    chart: kubernetes-dashboard-0.6.4
    release: kube-dashboard
    heritage: Tiller
  annotations:
    nginx.ingress.kubernetes.io/secure-backends: true
spec:
  rules:
    - host: my-host.tld
      http:
        paths:
          - path: /
            backend:
              serviceName: kube-dashboard-kubernetes-dashboard
              servicePort: 443
  tls:
    - hosts:
      - my-host.tld
      secretName: dashboard-tls-secret
```

In other words, when the string value happens to also be a valid representation of a boolean value, it's lacking the proper quoting.

**Comment by bacongobbler** [comment]
Copying what I responded in Slack:

Yes, that is completely intentional. `--set-string` does not quote values and as such you will still need to quote rendered values (as booleans or integers) as you would with `--set`. The only difference between `--set` and `--set-string` is that the latter avoids the problem where the yaml parser parses large integer values into scientific notation (#3001). See #3599 for more background context.

Closing this as working as intended. :)
